back/src/forms/database.ts

```diff
@@ -110,14 +120,47 @@ export function getFormSiretsByRole(
       form.forwardedIn?.recipientCompanySiret
     ].filter(Boolean),
     transportersSirets: [
-      form.transporterCompanySiret,
-      form.transporterCompanyVatNumber,
-      ...form.transportSegments?.map(
-        segment => segment.transporterCompanySiret
-      ),
-      form.forwardedIn?.transporterCompanySiret
+      ...(form.transporters ?? []).flatMap(t => [
+        t.transporterCompanySiret,
+        t.transporterCompanyVatNumber
+      ]),
+      ...(form.forwardedIn?.transporters ?? []).flatMap(t => [
+        t.transporterCompanySiret,
+        t.transporterCompanyVatNumber
+      ])
     ].filter(Boolean),
     intermediariesSirets:
       form.intermediaries?.map(intermediary => intermediary.siret) ?? []
   };
 }
+
+export async function getTransporters(
+  form: Pick<Form, "id">
+): Promise<BsddTransporter[]> {
+  const transporters = await prisma.form
+    .findUnique({ where: { id: form.id } })
+    .transporters({ orderBy: { number: "asc" } });
+  return transporters ?? [];
+}
+
+export function getTransportersSync(
+  form: Form & { transporters: BsddTransporter[] | null }
+): BsddTransporter[] {
+  return form.transporters ?? [];
+}
+
+export async function getFirstTransporter(
+  form: Pick<Form, "id">
+): Promise<BsddTransporter | null> {
+  const transporters = await getTransporters(form);
+  const [firstTransporter] = transporters.filter(t => t.number === 1);
+  return firstTransporter ?? null;
+}
+
+export function getFirstTransporterSync(
+  form: Form & { transporters: BsddTransporter[] | null }
+): BsddTransporter | null {
+  const transporters = getTransportersSync(form);
+  const [firstTransporter] = transporters.filter(t => t.number === 1);
+  return firstTransporter ?? null;
```

Comment:
Nitpick, mais éventuellement

```suggestion
 return transporters.find(t => t.number === 1);
```

A noter que ca retournerait undefined au lieu de null si ca ne trouve pas.

Reply:
En effet c'est mieux !